### PR TITLE
fix: encrypt attribute when not listed on `accept` list

### LINKED ISSUE #34: Handle encrypted attributes changed by arguments.
I have this change:

```elixir
defmodule Canary.Change.GenerateKey do
  use Ash.Resource.Change

  @impl true
  def init(opts) do
    if is_nil(opts[:output_attribute]) and is_nil(opts[:output_argument]) do
      :error
    else
      {:ok, opts}
    end
  end

  @impl true
  def atomic(changeset, opts, context) do
    {:ok, change(changeset, opts, context)}
  end

  @impl true
  def change(changeset, opts, _context) do
    length = opts[:length] || 32

    random_string =
      :crypto.strong_rand_bytes(length)
      |> Base.url_encode64()
      |> binary_part(0, length)

    if not is_nil(opts[:output_attribute]) do
      Ash.Changeset.force_change_attribute(
        changeset,
        opts[:output_attribute],
        random_string
      )
    else
      Ash.Changeset.force_set_argument(
        changeset,
        opts[:output_argument],
        random_string
      )
    end
  end
end
```

And it seems like current [AshCloak's implementation](https://github.com/ash-project/ash_cloak/blob/044a3d8eed2ce438ad154b3e016915194233a349/lib/ash_cloak/transformers/set_up_encryption.ex#L84-L122) do not support this. It only works with `accept` when I tested it.

```elixir
  attributes do
    uuid_primary_key :id

    attribute :value, :string, allow_nil?: false
  end

  identities do
    identity :unique_value, [:value]
  end

  relationships do
    belongs_to :account, Canary.Accounts.Account
  end

    create :create do
      primary? true

      argument :account_id, :uuid, allow_nil?: false
      change manage_relationship(:account_id, :account, type: :append)

      change {
        Canary.Change.GenerateKey,
        length: 32, output_attribute: :value
      }
    end
```

### PR DESCRIPTION
**Pull request by suciptoid**
### Contributor checklist

- [x] Bug fixes include regression tests
- [ ] Features include unit/acceptance tests

fix #34

### CODE REVIEW

File: lib/ash_cloak/transformers/set_up_encryption.ex [hunk 1]
```diff
@@ -82,11 +82,13 @@ defmodule AshCloak.Transformers.SetupEncryption do
...
 
   defp rewrite_actions({:ok, dsl}, attr) do
+    cloaked_attrs = AshCloak.Info.cloak_attributes!(dsl)
+
     dsl
     |> Ash.Resource.Info.actions()
     |> Enum.filter(&(&1.type in [:create, :update, :destroy]))
     |> Enum.reduce_while({:ok, dsl}, fn action, {:ok, dsl} ->
-      if attr.name in action.accept do
+      if attr.name in (action.accept ++ cloaked_attrs) do
```
[the remark this answers is not in the document]
Comment (zachdaniel) [reply lib/ash_cloak/transformers/set_up_encryption.ex:91]: We can add exceptions later if people want to write directly to the encrypted value I guess. The thing that worries me is that we have to be very very very very careful not to encrypt an already encrypted value. That is a data-loss situation.

**Comment by zachdaniel** [comment]
I was on board with this at first, but there are some hidden dangers with encrypting changes when we can't guarantee we aren't double encrypting. For now, `AshCloak.encrypt_and_set` is the only way. Perhaps a PR to the documentation explaining this?

**Comment by suciptoid** [comment]
I've same problem with #34 few month ago, and also fixed by including encrypted attributes to `accept`.

```elixir
    create :generate do
      accept [:name, :private_key, :public_key]

      change Servers.Key.Generate
    end
```

```elixir
defmodule MyApp.Servers.Key.Generate do
  use Ash.Resource.Change

  alias Ash.Changeset

  def change(changeset, _opts, _context) do
    case SSH.Key.gen(:rsa, "bot@local.dev") do
      {pub, key} ->
        changeset
        |> Changeset.force_change_attribute(:public_key, pub) # <-- non encrypted
        |> Changeset.force_set_argument(:private_key, key) # <-- encrypted
    end
  end
end
```

Since this isn't mentioned in the documentation, I initially expected that including only `:name` in the **accept** list would be sufficient (as the encrypted attribute is generated within `Resource.Change`). I assumed that `Changeset.force_set_argument(:private_key, key)` would still function as intended.


```elixir
    create :generate do
      accept [:name]

      change Servers.Key.Generate
    end
```

**Comment by zachdaniel** [comment]
🤔Hm...it's actually kind of interesting because that does still theoretically open you up to doubly encrypting values. So maybe your PR is actually correct, that the semantics should just be to always encrypt the value before write if its being set 🤔 

Can you reopen this PR? I will approve it. We just need to explain to folks somewhere in the docs not to ever set the attribute back to its encrypted value (or an encrypted value) or they risk double encryption.

**Comment by suciptoid** [comment]
If I understand correctly, it's already handled by ash_cloak (?). The cloaked attribute has been converted to an argument, and it is no longer possible to set or change the attribute.

```elixir
|> Changeset.force_change_attribute(:public_key, pub) # <-- non encrypted
|> Changeset.force_set_argument(:private_key, key) # <-- encrypted, use force_set_argument not force_change_attribute
```

**Comment by suciptoid** [comment]
I can't reopen this PR, so I open another PR #46 with your change suggestion.

**Pull request closed without merging**